src/aws-cpp-sdk-core/include/smithy/interceptor/InterceptorContext.h

```diff
@@ -17,78 +17,57 @@ namespace smithy
-                }
-                return m_request;
+                return m_transmitRequest;
             }
 
-            void SetRequest(const std::shared_ptr<Aws::Http::HttpRequest>& request)
+            void SetTransmitRequest(const std::shared_ptr<Aws::Http::HttpRequest>& transmitRequest)
             {
-                this->m_request = request;
+                m_transmitRequest = transmitRequest;
             }
 
-            using GetResponseOutcome = Aws::Utils::Outcome<std::shared_ptr<Aws::Http::HttpResponse>, Aws::Client::AWSError<Aws::Client::CoreErrors>>;
-            GetResponseOutcome GetResponse() const
+            std::shared_ptr<Aws::Http::HttpResponse> GetTransmitResponse() const
             {
-                if (!m_response)
-                {
-                    return Aws::Client::AWSError<Aws::Client::CoreErrors>{
-                        Aws::Client::CoreErrors::RESOURCE_NOT_FOUND,
-                        "ResourceNotFoundException",
-                        "Response is NULL",
-                        false
-                    };
-                }
-                return m_response;
+                return m_transmitResponse;
             }
 
-            void SetResponse(const std::shared_ptr<Aws::Http::HttpResponse>& response)
+            void SetTransmitResponse(const std::shared_ptr<Aws::Http::HttpResponse>& transmitResponse)
             {
-                this->m_response = response;
+                m_transmitResponse = transmitResponse;
             }
 
-            using GetAttributeOutcome = Aws::Utils::Outcome<Aws::String, Aws::Client::AWSError<Aws::Client::CoreErrors>>;
-            GetAttributeOutcome GetAttribute(const Aws::String& attribute)
+            Aws::String GetAttribute(const Aws::String& key) const
             {
-                const auto attribute_iter = m_attributes.find(attribute);
-                if (attribute_iter == m_attributes.end())
-                {
-                    return Aws::Client::AWSError<Aws::Client::CoreErrors>{
-                        Aws::Client::CoreErrors::RESOURCE_NOT_FOUND,
-                        "ResourceNotFoundException",
-                        "Attribute not found",
-                        false
-                    };
-                }
-                return attribute_iter->second;
+                return m_attributes.at(key);
             }
 
-            void SetAttribute(const Aws::String& attribute, const Aws::String& value)
+            void SetAttribute(const Aws::String& key, const Aws::String& value)
             {
-                m_attributes.emplace(attribute, value);
+                m_attributes.insert({key, value});
             }
 
         private:
             Aws::Map<Aws::String, Aws::String> m_attributes{};
-            std::shared_ptr<Aws::Http::HttpRequest> m_request{nullptr};
-            std::shared_ptr<Aws::Http::HttpResponse> m_response{nullptr};
+            const Aws::AmazonWebServiceRequest& m_modeledRequest;
```

Comment:
storing reference? what happens if caller out of scope. is that possible in this workflow?

Reply:
nope its a borrow from the parent scope, in the current workflow its not possible